pkg/controller/ps/suite_test.go

```diff
@@ -77,32 +74,57 @@ var _ = BeforeSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 
-	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
-		Scheme: scheme.Scheme,
-	})
-	Expect(err).ToNot(HaveOccurred())
-
-	err = (&PerconaServerMySQLReconciler{
-		Client: k8sManager.GetClient(),
-		Scheme: k8sManager.GetScheme(),
-		ServerVersion: &platform.ServerVersion{
-			Platform: platform.PlatformKubernetes,
-		},
-	}).SetupWithManager(k8sManager)
+	err = cmscheme.AddToScheme(k8sClient.Scheme())
 	Expect(err).ToNot(HaveOccurred())
-
-	//+kubebuilder:scaffold:scheme
-
-	go func() {
-		defer GinkgoRecover()
-		err = k8sManager.Start(ctx)
-		Expect(err).ToNot(HaveOccurred(), "failed to run manager")
-	}()
 })
 
 var _ = AfterSuite(func() {
-	cancel()
 	By("tearing down the test environment")
 	err := testEnv.Stop()
 	Expect(err).NotTo(HaveOccurred())
 })
+
+func reconciler() *PerconaServerMySQLReconciler {
+	return (&PerconaServerMySQLReconciler{
+		Client: k8sClient,
+		Scheme: k8sClient.Scheme(),
+		ServerVersion: &platform.ServerVersion{
+			Platform: platform.PlatformKubernetes,
+		},
+	})
+}
+
+func readDefaultCR(name, namespace string) (*psv1alpha1.PerconaServerMySQL, error) {
+	sch := runtime.NewScheme()
+	err := scheme.AddToScheme(sch)
+	if err != nil {
+		return nil, err
+	}
+	err = psv1alpha1.AddToScheme(sch)
+	if err != nil {
+		return nil, err
+	}
+
+	decode := serializer.NewCodecFactory(sch).UniversalDeserializer().Decode
+
+	data, err := os.ReadFile(filepath.Join("..", "..", "..", "deploy", "cr.yaml"))
+	if err != nil {
+		return nil, err
+	}
+
+	obj, gKV, err := decode(data, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var cr *psv1alpha1.PerconaServerMySQL
+
+	if gKV.Kind == "PerconaServerMySQL" {
+		cr = obj.(*psv1alpha1.PerconaServerMySQL)
+	}
```

Comment:
I think we can simplify it by using `Unmarshal` from [k8s.io/apimachinery/pkg/util/yaml](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/yaml#Unmarshal)

Reply:
I saw that in your PR, I think you're right.

Reply:
Done: [ff4f892](https://github.com/percona/percona-server-mysql-operator/pull/314/commits/ff4f892af6bda9bea2a830b17e6e809302a4e249)